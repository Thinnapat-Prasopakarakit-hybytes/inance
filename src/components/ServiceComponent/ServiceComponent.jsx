import React from "react";
import "./ServiceComponent.scss";
import s1 from "../../assets/images/s1.png";
import s2 from "../../assets/images/s2.png";
import s3 from "../../assets/images/s3.png";
import { Link } from "react-router-dom";
import { useIntl } from "react-intl";
const ServiceComponent = () => {
  const { messages, locale } = useIntl();
  const services = [
    {
      image: s1,
      title: messages.services.maintenance,
      description: messages.services.description,
    },
    {
      image: s2,
      title: messages.services.electrical,
      description: messages.services.description,
    },
    {
      image: s3,
      title: messages.services.plumbing,
      description: messages.services.description,
    },
  ];

  return (
    <section className="service_section layout_padding">
      <div className="container">
        <div className="heading_container heading_center">
          <h2>{messages.services.title}</h2>
        </div>
        <div className="row">
          {services.map((service, idx) => (
            <div className="col-sm-6 col-md-4 mx-auto px-3" key={idx}>
              <div className="box">
                <div className="img-box">
                  <img src={service.image} alt={service.title} />
                </div>
                <div className="detail-box">
                  <h5>{service.title}</h5>
                  <p>{service.description}</p>
                </div>
              </div>
            </div>
          ))}
        </div>
        <div className="btn-box">
          <Link to={`/${locale}/about`}> {messages.services.viewMore} </Link>
        </div>
      </div>
    </section>
  );
};

export default ServiceComponent;
