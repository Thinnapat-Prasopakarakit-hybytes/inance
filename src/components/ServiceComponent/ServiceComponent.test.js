import React from "react";
import { render, screen } from "@testing-library/react";
import ServiceComponent from "./ServiceComponent";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";

jest.mock("../../assets/images/s1.png", () => "mock-s1-image");
jest.mock("../../assets/images/s2.png", () => "mock-s2-image");
jest.mock("../../assets/images/s3.png", () => "mock-s3-image");

const renderServiceComponent = () => {
  return render(
    <BrowserRouter>
      <ServiceComponent />
    </BrowserRouter>
  );
};

describe("ServiceComponent", () => {
  it("should render the main heading", () => {
    renderServiceComponent();
    expect(screen.getByText("Our Services")).toBeInTheDocument();
  });

  it("should render all service items", () => {
    renderServiceComponent();

    const serviceTitles = screen.getAllByText(
      /Maintenance|Electrical|Plumbing/
    );
    expect(serviceTitles).toHaveLength(3);

    const serviceDescriptions = screen.getAllByText(
      /when looking at its layout/i
    );
    expect(serviceDescriptions).toHaveLength(3);

    const serviceImages = screen.getAllByRole("img");
    const expectedImages = [
      { src: "mock-s1-image", alt: "Maintenance" },
      { src: "mock-s2-image", alt: "Electrical" },
      { src: "mock-s3-image", alt: "Plumbing" },
    ];

    serviceImages.forEach((img, index) => {
      expect(img).toHaveAttribute("src", expectedImages[index].src);
      expect(img).toHaveAttribute("alt", expectedImages[index].alt);
    });
  });

  it("should render the 'View More' link", () => {
    renderServiceComponent();
    const viewMoreLink = screen.getByRole("link", { name: /View More/i });
    expect(viewMoreLink).toBeInTheDocument();
    expect(viewMoreLink).toHaveAttribute("href", "/");
  });

  it("should match the snapshot", () => {
    const { asFragment } = renderServiceComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render service images with correct sources", () => {
    renderServiceComponent();
    const serviceImages = screen.getAllByRole("img");

    serviceImages.forEach((img, index) => {
      console.log(`Image ${index}:`, {
        src: img.getAttribute("src"),
        alt: img.getAttribute("alt"),
      });
    });

    expect(serviceImages[0]).toHaveAttribute("src", "mock-s1-image");
    expect(serviceImages[1]).toHaveAttribute("src", "mock-s2-image");
    expect(serviceImages[2]).toHaveAttribute("src", "mock-s3-image");
  });
});
