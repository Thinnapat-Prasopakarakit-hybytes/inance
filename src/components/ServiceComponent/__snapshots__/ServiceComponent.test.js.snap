// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceComponent should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="service_section layout_padding"
  >
    <div
      class="container"
    >
      <div
        class="heading_container heading_center"
      >
        <h2>
          Our Services
        </h2>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-6 col-md-4 mx-auto px-3"
        >
          <div
            class="box"
          >
            <div
              class="img-box"
            >
              <img
                alt="Maintenance"
                src="mock-s1-image"
              />
            </div>
            <div
              class="detail-box"
            >
              <h5>
                Maintenance
              </h5>
              <p>
                when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-sm-6 col-md-4 mx-auto px-3"
        >
          <div
            class="box"
          >
            <div
              class="img-box"
            >
              <img
                alt="Electrical"
                src="mock-s2-image"
              />
            </div>
            <div
              class="detail-box"
            >
              <h5>
                Electrical
              </h5>
              <p>
                when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-sm-6 col-md-4 mx-auto px-3"
        >
          <div
            class="box"
          >
            <div
              class="img-box"
            >
              <img
                alt="Plumbing"
                src="mock-s3-image"
              />
            </div>
            <div
              class="detail-box"
            >
              <h5>
                Plumbing
              </h5>
              <p>
                when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="btn-box"
      >
        <a
          href="/"
        >
           View More 
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
